src/EnergyPlus/ReportCoilSelection.cc

```diff
@@ -1504,16 +1500,17 @@ void ReportCoilSelection::setCoilHeatingCapacity(
     c->zoneEqNum = curZoneEqNum;
     //    if ( c->zoneEqNum > 0 ) doZoneEqSetup( index );
     if (curSysNum > 0 && c->zoneEqNum == 0 && allocated(state.dataSize->FinalSysSizing)) {
-        c->desDayNameAtSensPeak = state.dataSize->FinalSysSizing(curSysNum).HeatDesDay;
-
-        c->oaPeakTemp = state.dataSize->FinalSysSizing(curSysNum).HeatOutTemp;
-        c->oaPeakHumRat = state.dataSize->FinalSysSizing(curSysNum).HeatOutHumRat;
-        c->oaPeakVolFlow = state.dataSize->FinalSysSizing(curSysNum).DesOutAirVolFlow;
-        c->raPeakTemp = state.dataSize->FinalSysSizing(curSysNum).HeatRetTemp;
-        c->raPeakHumRat = state.dataSize->FinalSysSizing(curSysNum).HeatRetHumRat;
-        c->coilSizingMethodConcurrence = state.dataSize->FinalSysSizing(curSysNum).SizingOption;
-        c->coilSizingMethodCapacity = state.dataSize->FinalSysSizing(curSysNum).HeatingCapMethod;
-        c->coilSizingMethodAirFlow = state.dataSize->FinalSysSizing(curSysNum).ScaleHeatSAFMethod;
+        auto finalSysSizing = state.dataSize->FinalSysSizing(curSysNum);
```

Comment:
This needs to be an `auto &` (reference), otherwise you are making a local copy of the object.